MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -0,0 +1,275 @@
+import Polyline
+import MapboxDirections
+import AVFoundation
+import MapboxMobileEvents
+
+let SecondsBeforeCollectionAfterFeedbackEvent: TimeInterval = 20
+
+struct EventDetails {
+    var originalRequestIdentifier: String?
+    var requestIdentifier: String?
+    var coordinate: CLLocationCoordinate2D?
+    var originalGeometry: Polyline?
+    var originalDistance: CLLocationDistance?
+    var originalEstimatedDuration: TimeInterval?
+    var originalStepCount: Int?
+    var geometry: Polyline?
+    var distance: CLLocationDistance?
+    var estimatedDuration: TimeInterval?
+    var stepCount: Int?
+    var created: Date
+    var startTimestamp: String?
+    var platform: String
+    var operatingSystem: String
+    var device: String
+    var sdkIdentifier: String
+    var sdkVersion: String
+    var eventVersion: Int
+    var profile: String
+    var simulation: Bool
+    var sessionIdentifier: String
+    var distanceCompleted: CLLocationDistance
+    var distanceRemaining: TimeInterval
+    var durationRemaining: TimeInterval
+    var rerouteCount: Int
+    var volumeLevel: Int
+    var screenBrightness: Int
+    var batteryPluggedIn: Bool
+    var batteryLevel: Float
+    var applicationState: String
```

Comment:
This should be a UIApplicationState. Wait until `convertedToDictionary()` to lazily convert it to a string.